### PR TITLE
Rework event form structure  #5895

### PR DESCRIPTION
**Pull request by phpgeekfr**

### CODE REVIEW

File: apps/festival/festival/src/app/dashboard/dashboard.module.ts [hunk 1]
```diff
@@ -83,10 +99,59 @@ const routes: Routes = [{
...
+              component: ReviewComponent,
+              children: [
+                {
+                  path: '',
+                  pathMatch: 'full',
+                  redirectTo: 'statistics'
+                },
+                {
+                  path: 'invitations',
+                  component: GuestListComponent,
```
Comment (GrandSchtroumpf) [inline apps/festival/festival/src/app/dashboard/dashboard.module.ts:111]: use loadChildren for all the children components.

File: libs/event/src/lib/form/meeting/meeting.component.scss [hunk 2]
```diff
@@ -0,0 +1,13 @@
+.link {
+  mat-form-field {
+    display: block;
+  }
+
+  button {
+    margin-top: 20px;
+  }
+
+  .mat-form-field-appearance-outline.mat-form-field-disabled .mat-form-field-outline {
```
Comment (GrandSchtroumpf) [inline libs/event/src/lib/form/meeting/meeting.component.scss:10]: Looks like somethink that should either be in the theme ?

File: libs/event/src/lib/form/screening/screening.component.scss [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+.link {
+  mat-form-field {
+    display: block;
+  }
+
+  button {
+    margin-top: 20px;
+  }
+
+  .mat-form-field-appearance-outline.mat-form-field-disabled .mat-form-field-outline {
```
Comment (GrandSchtroumpf) [inline libs/event/src/lib/form/screening/screening.component.scss:11]: same as above

File: libs/event/src/lib/form/shell/shell.component.html [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+        Save & Exit</button>
+      <button class="text-primary" mat-button [mat-dialog-close]="false">
+        Close without saving
+      </button>
+    </div>
+  </div>
+</ng-template>
+
+<form [formGroup]="form">
+  <section fxLayout="column" fxLayoutGap="24px">
```
Comment (GrandSchtroumpf) [inline libs/event/src/lib/form/shell/shell.component.html:33]: For me this is not necessary. `nav` is a block, and `mat-card` too I think. If not better set them as `block` than adding a parent with flexbox.

File: libs/event/src/lib/form/shell/shell.component.html [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+      </ng-container>
+    </nav>
+  </section>
+</form>
+
+<div class="routerContainer" [@routeAnimation]="animationOutlet(outlet)">
+  <router-outlet #outlet="outlet"></router-outlet> 
+</div>
+
+<div class="buttonContainer" fxLayoutAlign="center">
```
Comment (GrandSchtroumpf) [inline libs/event/src/lib/form/shell/shell.component.html:56]: why do you need a container ?

File: apps/festival/festival/src/app/dashboard/event/review/review.component.html [hunk 6]
```diff
@@ -1,4 +1,2 @@
-<ng-container *ngIf="event$ | async as event">
-  <event-review [event]="event"></event-review>
-  <event-analytics [event]="event"></event-analytics>
-</ng-container>
+<event-review></event-review>
+
```
[the remark this answers is not in the document]
Comment (phpgeekfr) [reply apps/festival/festival/src/app/dashboard/event/review/review.component.html]: This is already the case, you looked at an outdated version of the PR

File: libs/event/src/lib/form/invitation/invitation.component.html [hunk 7]
```diff
@@ -0,0 +1,5 @@
+﻿<section class="invitations surface" fxLayout="column">
```
[the remark this answers is not in the document]
Comment (phpgeekfr) [reply libs/event/src/lib/form/invitation/invitation.component.html:1]: How can I handle the `fxLayout="column"` in that case ?

For many classes, can I :
```
host: {
    class: 'surface invitations'
  },
```
 ?

File: libs/event/src/lib/form/meeting-files/meeting-files.component.html [hunk 8]
```diff
@@ -0,0 +1,33 @@
+﻿<section class="surface">
```
[the remark this answers is not in the document]
Comment (phpgeekfr) [reply libs/event/src/lib/form/meeting-files/meeting-files.component.html]: In fact, this is not working as intended, I would like to show you the changes

File: libs/event/src/lib/form/shell/shell.component.html (hunk 5 above)
Comment (phpgeekfr) [reply libs/event/src/lib/form/shell/shell.component.html]: I managed to remove the fxLayout in favor of small css update but Antoine and I did not found any way to remove container for expected results

File: libs/event/src/lib/form/meeting/meeting.component.scss (hunk 2 above)
Comment (phpgeekfr) [reply libs/event/src/lib/form/meeting/meeting.component.scss:10]: @Poppe-A will ask @stephanie-cascade8 on this particular point.

File: libs/event/src/lib/form/shell/shell.component.scss [hunk 9]
```diff
@@ -0,0 +1,23 @@
...
+    padding: 24px;
+    margin-bottom: 24px;
+  }
+  nav {
+    border-bottom: none;
+  }
+}
+
+.routerContainer {
+  margin: 16px 16px 0 16px;
```
Comment (GrandSchtroumpf) [inline libs/event/src/lib/form/shell/shell.component.scss:17]: just put 16px everywhere

File: libs/event/src/lib/form/shell/shell.component.ts [hunk 10]
```diff
@@ -0,0 +1,141 @@
...
+    { path: 'invitations', label: 'Invitations' },
+    { path: 'files', label: 'Files' },
+  ],
+}
+@Component({
+  selector: 'event-shell',
+  templateUrl: './shell.component.html',
+  styleUrls: ['./shell.component.scss'],
+  animations: [routeAnimation],
+  changeDetection: ChangeDetectionStrategy.Default  // required for changes on "pristine" for the save button
```
Comment (GrandSchtroumpf) [inline libs/event/src/lib/form/shell/shell.component.ts:32]: I think that putting `markForCheck` after the pristine should help.

File: libs/event/src/lib/form/shell/shell.component.ts [hunk 11]
```diff
@@ -0,0 +1,141 @@
...
+      this.form = new EventForm(event);
+
+      const type = this.form.value.type;
+      const path = type === 'meeting' ? 'lobby' : 'session';
+      this.internalLink = `/c/o/marketplace/event/${this.form.value.id}/${path}`;
+      const app = getCurrentApp(this.routerQuery);
+      const url = applicationUrl[app];
+      this.link = `${url}${this.internalLink}`;
+
+      this.tabs$ = this.service.valueChanges(this.form.value.id).pipe(
```
Comment (GrandSchtroumpf) [inline libs/event/src/lib/form/shell/shell.component.ts:68]: avoid subcriptions in subscriptions. This is a super dangerous.

File: styles/components/mat-form-field.scss [hunk 12]
```diff
@@ -73,5 +73,10 @@
         border-width: 1px;
       }
     }
+    &.mat-form-field-disabled {
```
Comment (GrandSchtroumpf) [inline styles/components/mat-form-field.scss:76]: did you check with Stéphanie ?

**Review by GrandSchtroumpf — CHANGES_REQUESTED** [review]

**Review by GrandSchtroumpf — APPROVED** [review]

File: libs/event/src/lib/form/shell/shell.component.ts (hunk 11 above)
Comment (phpgeekfr) [reply libs/event/src/lib/form/shell/shell.component.ts:68]: created #6780

File: styles/components/mat-form-field.scss (hunk 12 above)
Comment (phpgeekfr) [reply styles/components/mat-form-field.scss:76]: yes  @Poppe-A did

File: libs/event/src/lib/form/meeting-files/meeting-files.component.html [hunk 13]
```diff
@@ -0,0 +1,30 @@
...
+    <span>Add Files</span>
+  </button>
+</div>
+
+<p>Selected Files ({{files.value.length}})</p>
+<mat-divider></mat-divider>
+
+<ng-container *ngIf="!!files.value.length; else empty">
+  <mat-list>
+    <mat-list-item *ngFor="let file of files.value; let i = index" (click)="previewFile(file)">
```
[the remark this answers is not in the document]
Comment (phpgeekfr) [reply libs/event/src/lib/form/meeting-files/meeting-files.component.html:19]: From the doc, mat-action-list should contain only a button. this is not the case here

**Pull request merged**